### PR TITLE
For #6070 Adds telemetry for enabling search suggestions in private

### PR DESCRIPTION
**Pull request by mcarare**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by mcarare** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) **What questions will you answer with this data?**

- How often users choose to enable receiving search suggestions in private sessions.

2) **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**

- We want to know how users perceive search suggestions as regarding private browsing. This will help us choose the default setting for search suggestions in private - enabled or disabled.

3) **What alternative methods did you consider to answer these questions? Why were they not sufficient?**

* No current alternative methods.


4) **Can current instrumentation answer these questions?**
* Currently there are no events to track this, this is a new feature.

5) **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data c](https://wiki.mozilla.org/Firefox/Data_Collection)[ollection ](https://wiki.mozilla.org/Firefox/Data_Collection)[categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the found on the Mozilla wiki.**

* All data is Category 2.

6) **How long will this data be collected?**

* Until 03/01/2020

7) **What populations will you measure?**

* All release, beta, and nightly users with telemetry enabled.

8) **Please provide a general description of how you will analyze this data.**

* Glean / Amplitude

9) **Where do you intend to share the results of your analysis?**

* Only on Glean, Amplitude and with mobile teams.

File: app/metrics.yaml [hunk 1]
```diff
@@ -114,22 +114,22 @@ events:
         description: "The preference key for the boolean (true/false) preference the user toggled. We currently track:
           show_search_suggestions, remote_debugging, telemetry, tracking_protection, search_bookmarks,
           search_browsing_history, show_clipboard_suggestions, show_search_shortcuts, open_links_in_a_private_tab,
-          pref_key_sync_logins, pref_key_sync_bookmarks and pref_key_sync_history"
```
Comment (boek) [inline app/metrics.yaml]: Did you intend to remove these two items?

File: app/metrics.yaml [hunk 2]
```diff
@@ -114,22 +114,22 @@ events:
...
+          pref_key_sync_logins and pref_key_show_search_suggestions_in_private"
       enabled:
         description: "Whether or not the preference is *now* enabled"
     bugs:
     - https://github.com/mozilla-mobile/fenix/issues/975
     - https://github.com/mozilla-mobile/fenix/issues/5094
     - https://github.com/mozilla-mobile/fenix/issues/5737
     - https://github.com/mozilla-mobile/fenix/issues/5586
-    - https://github.com/mozilla-mobile/fenix/issues/6396
```
Comment (boek) [inline app/metrics.yaml]: Did you intend to rename this? Maybe this was a bad merge? :)

**Comment by boek** [comment]
# Data Review Form (to be filled by Data Stewards)

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?
Yes, metrics.yaml and metrics.md

2) Is there a control mechanism that allows the user to turn the data collection on and off?
Yes, under data controls

3) If the request is for permanent data collection, is there someone who will monitor the data over time?
Has expiry

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?
Type 2, Type 1

5) Is the data collection request for default-on or default-off?
Default on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?
No

7) Is the data collection covered by the existing Firefox privacy notice?
Yes

8) Does there need to be a check-in in the future to determine whether to renew the data?
Will check-in at expiry

9) Does the data collection use a third-party collection tool?
No

**Comment by boek** [comment]
Also looks like tests are failing?

File: app/metrics.yaml (hunk 2 above)
Comment (mcarare) [reply app/metrics.yaml]: Yes, there was a strange merge...

**Pull request merged**